runtime/queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,308 @@
+(identifier) @variable
+; Assume all-caps names are constants
+((identifier) @constant
+  (#lua-match? @constant "^[A-Z][A-Z0-9_]+$"))
+
+; Function definitions/declarations
+(function_definition
+  name: (identifier) @function)
+
+(function_declaration
+  name: (identifier) @function)
+
+(parameter_declaration
+  name: (identifier) @variable.parameter)
+
+; Methods / Properties
+(field_access
+  field: (identifier) @variable.member)
+
+; Function calls
+(call_expression
+  function: (identifier) @function)
+
+(call_expression
+  function:
+    (field_access
+      field: (identifier) @function.method.call)) ; Must be after field_access
+
+; Types
+(builtin_type) @type.builtin
+
+(type
+  (identifier) @type)
+
+(any_type) @type
+
+; Variables
+(variable_storage_class) @keyword.storage
+
+(variable_declaration
+  name: (identifier) @variable)
+
+(old_variable_declaration
+  name: (identifier) @variable)
+
+; Preprocessor
+(preproc_include) @keyword.import
+
+(preproc_tryinclude) @keyword.import
+
+(system_lib_string) @string
+
+(string_literal) @string
+
+(preproc_assert) @keyword.directive
+
+(preproc_pragma) @keyword.directive
+
+(preproc_arg) @constant
+
+(preproc_macro) @function.macro
+
+(macro_param) @variable.parameter
+
+(preproc_if) @keyword.directive
+
+(preproc_else) @keyword.directive
+
+(preproc_elseif) @keyword.directive
+
+(preproc_endif) @keyword.directive
+
+(preproc_endinput) @keyword.directive
+
+(preproc_define) @keyword.directive.define
+
+(preproc_define
+  name: (identifier) @constant)
+
+(preproc_undefine) @keyword.directive.define
+
+(preproc_undefine
+  name: (identifier) @constant)
+
+(preproc_error) @function.macro
+
+(preproc_warning) @function.macro
+
+; Expressions
+(view_as) @function.builtin
+
+(sizeof_expression) @function.macro
+
+(this) @variable.builtin
+
+; https://github.com/alliedmodders/sourcemod/blob/5c0ae11a4619e9cba93478683c7737253ea93ba6/plugins/include/handles.inc#L78
+(hardcoded_symbol) @variable.builtin
+
+; Comments
+(comment) @comment
+
+; General
+(parameter_declaration
+  defaultValue: (identifier) @constant)
+
+(fixed_dimension) @punctuation.bracket
+
+(dimension) @punctuation.bracket
+
+(array_indexed_access) @punctuation.bracket
+
+(escape_sequence) @string.escape
```

Comment:
```suggestion
(escape_sequence) @constant.character.escape
```

the captures should line up with https://docs.helix-editor.com/master/themes.html#syntax-highlighting